File: 682. Baseball Game/Java/Solution.java
class Solution {
    public int calPoints(String[] ops) {
        Stack<Integer> history = new Stack<>();
        for (String op : ops) {
            switch (op) {
            case "+":
                history.push(history.peek() + history.get(history.size() - 2));
                break;
            case "D":
                history.push(history.peek() * 2);
                break;
            case "C":
                history.pop();
                break;
            default:
                history.push(Integer.parseInt(op));
            }
        }
        int result = 0;
        for (int item : history) {
            result += item;
        }
        return result;
    }
}

File: 682. Baseball Game/C++/Solution.cpp
class Solution {
public:
    int calPoints(vector<string> &ops) {
        vector<int> history;
        for (auto &op : ops) {
            if (op == "+") {
                history.push_back(history.back() + history[history.size() - 2]);
            } else if (op == "D") {

                history.push_back(history.back() * 2);
            } else if (op == "C") {
                history.pop_back();
            } else {
                history.push_back(stoi(op));
            }
        }
        return accumulate(history.begin(), history.end(), 0);
    }
};